src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java

```diff
@@ -0,0 +1,258 @@
+
+    serversPanel = new JPanel(new BorderLayout());
+
+    ToolbarDecorator toolbarDecorator = ToolbarDecorator.createDecorator(serverList).disableUpDownActions();
+
+    toolbarDecorator.setAddAction(new AnActionButtonRunnable() {
+      @Override
+      public void run(AnActionButton anActionButton) {
+        DefaultActionGroup group = new DefaultActionGroup();
+        group.add(new AddServerAction());
+
+        JBPopupFactory.getInstance()
+          .createActionGroupPopup(LABEL_ADD_SERVER, group, DataManager.getInstance().getDataContext(anActionButton.getContextComponent()),
+            JBPopupFactory.ActionSelectionAid.SPEEDSEARCH, true).show(anActionButton.getPreferredPopupPoint());
+      }
+    });
+
+    toolbarDecorator.setRemoveAction(new AnActionButtonRunnable() {
+      @Override
+      public void run(AnActionButton anActionButton) {
+        SonarQubeServer server = getSelectedServer();
+        if (server == null) {
+          return;
+        }
+
+        CollectionListModel model = (CollectionListModel) serverList.getModel();
+        // it's not removed from serverIds and editorList
+        model.remove(server);
+        servers.remove(server);
+
+        if (model.getSize() > 0) {
+          serverList.setSelectedValue(model.getElementAt(0), true);
+        }
+        else {
+          ((CardLayout) editorPanel.getLayout()).show(editorPanel, EMPTY_CARD_NAME);
+        }
+      }
+    });
+
+    JBLabel serversLabel = new JBLabel("Configured servers:");
+    serversLabel.setLabelFor(serverList);
+    serversPanel.add(serversLabel, BorderLayout.NORTH);
+    serversPanel.add(toolbarDecorator.createPanel(), BorderLayout.CENTER);
+
+    editorPanel = new JPanel(new CardLayout());
+
+    splitter = new Splitter(true);
+    splitter.setFirstComponent(serversPanel);
+    splitter.setSecondComponent(editorPanel);
+
+    JBLabel emptyLabel = new JBLabel("No server selected", SwingConstants.CENTER);
+    emptyPanel = new JPanel(new BorderLayout());
+    emptyPanel.add(emptyLabel, BorderLayout.CENTER);
+
+    panel = new JPanel(new BorderLayout());
+    panel.add(splitter);
+
+    serverList.getSelectionModel().addListSelectionListener(new ListSelectionListener() {
+      public void valueChanged(@NotNull ListSelectionEvent e) {
+        SonarQubeServer server = getSelectedServer();
+        if (server != null) {
+          String name = serverIds.get(server);
+          ((CardLayout) editorPanel.getLayout()).show(editorPanel, name);
+          splitter.doLayout();
+          splitter.repaint();
+        }
+      }
+    });
+
+    serverList.setCellRenderer(new ColoredListCellRenderer() {
+      @Override
+      protected void customizeCellRenderer(JList list, Object value, int index, boolean selected, boolean hasFocus) {
+        SonarQubeServer server = (SonarQubeServer)value;
+        try {
+          setIcon(ResourceLoader.getIcon(LIST_ICON));
+        } catch (IOException e) {
+          e.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1148)